src/d3/index.js

```diff
@@ -508,6 +508,36 @@ export const generateGraph = (
       getArrowEndColor(source, target, pipelineColors, colorMap)
     );
 
+    // Highlight blocked and blocking issues on hover.
+    nodes
+      .on( 'mouseenter', ( _e, { data } ) => {
+        const { id, parentIds } = data;
+  
+        nodes
+          .filter( ( d ) =>
+              ! [ ...parentIds, id ].includes( d.data.id ) &&
+              ! d.data.parentIds.includes( id )
+          )
+          .attr( 'opacity', '0.2' );
```

Comment:
> Personally, I find this opacity a _little_ severe in use. Can we bump it to 0.3?

I've bumped up the opacity to 0.3, thank you!

> What would be really nice is if we could do a quick fade, not sure how easy that would be...

I've also added a transition, please let me know if that looks good to you. I tried to add the transition to the DOM, but that didn't work, so I had to add this to the stylesheet.

> OTOH, it does strike me it could be an improvement to show an outline around the related issues rather than make them opaque, I think that could feel a bit "smoother"... We could then show different coloured outlined for blocked vs blocking issues too... Maybe one for a future iteration :)

I actually tried that initially but it felt that the opacity provided an additional level of visual differentiation from other issues. Also, there is an existing outline for the selected sprint, which I think could also lead to some confusion. What do you think?

Reply:
Thanks @nfmohit!

> > What would be really nice is if we could do a quick fade, not sure how easy that would be...
> 
> I've also added a transition, please let me know if that looks good to you. I tried to add the transition to the DOM, but that didn't work, so I had to add this to the stylesheet.

Adding it to the stylesheet seems fine and the transition LGTM!

> > OTOH, it does strike me it could be an improvement to show an outline around the related issues rather than make them opaque, I think that could feel a bit "smoother"... We could then show different coloured outlined for blocked vs blocking issues too... Maybe one for a future iteration :)
> 
> I actually tried that initially but it felt that the opacity provided an additional level of visual differentiation from other issues.

I'd have to see it in action to really appreciate the difference, but I can imagine what you mean.

> Also, there is an existing outline for the selected sprint, which I think could also lead to some confusion. What do you think?

I don't see this being _too_ much of an issue depending on how we styled the new outline. There's already a scenario where the drop target outline is being drawn outside of the selected sprint outline; admittedly it's not perfect aesthetically, but it works.

https://github.com/techanvil/zenhub-dependency-graph/blob/4f8a63919f89812ec641dc425ed685cee701d449/src/d3/select-and-drag.js#L134-L137

Anyhow, it's something I'd like to have a play with some time, just to try it out at least but I'm happy to stick with the current approach for now if not the foreseeable!